### PR TITLE
cgroup: enable memory swappiness equal to -1

### PR DESCRIPTION
**Pull request by Ace-Tang**
Signed-off-by: Ace-Tang <aceapril@126.com>

### Ⅰ. Describe what this PR did

enable memory-swappiness can be set to -1, -1 means do nothing.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

add！

### CODE REVIEW

File: apis/opts/memory_swappiness.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import "fmt"
 
 // ValidateMemorySwappiness verifies the correctness of memory-swappiness.
 func ValidateMemorySwappiness(memorySwappiness int64) error {
-	if memorySwappiness < 0 || memorySwappiness > 100 {
+	if memorySwappiness != -1 && (memorySwappiness < 0 || memorySwappiness > 100) {
```
Comment (HusterWan) [inline apis/opts/memory_swappiness.go]: how about `if  (memorySwappiness < -1 || memorySwappiness > 100)`

File: daemon/mgr/container_validation.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func validateResource(r *types.Resources, update bool) ([]string, error) {
 			warnings = append(warnings, MemorySwappinessWarn)
 			r.MemorySwappiness = nil
 		}
-		if r.MemorySwappiness != nil && (*r.MemorySwappiness < 0 || *r.MemorySwappiness > 100) {
+		if r.MemorySwappiness != nil && *r.MemorySwappiness != -1 && (*r.MemorySwappiness < 0 || *r.MemorySwappiness > 100) {
```
Comment (HusterWan) [inline daemon/mgr/container_validation.go]: how about `if r.MemorySwappiness != nil  (*r.MemorySwappiness < -1 || *r.MemorySwappiness > 100)`

File: apis/opts/memory_swappiness.go (hunk 1 above)
Comment (Ace-Tang) [reply apis/opts/memory_swappiness.go]: Do not do this, [0, 100] is a valid value, -1 is the value we enable to follow with moby.

Comment (HusterWan) [reply apis/opts/memory_swappiness.go]: OK, You are the boss :)

**Comment by HusterWan** [comment]
LGTM

**Pull request merged**